master/internal/api_runs.go

```diff
@@ -239,3 +251,201 @@ func sortRuns(sortString *string, runQuery *bun.SelectQuery) error {
+			if res.Error != nil {
+				failedExpMoveIds = append(failedExpMoveIds, res.ID)
+			}
+		}
+		var acceptedIDs []int32
+		if _, err = tx.NewUpdate().Table("runs").
+			Set("project_id = ?", req.DestinationProjectId).
+			Where("runs.id IN (?)", bun.In(validIDs)).
+			Where("runs.experiment_id NOT IN (?)", bun.In(failedExpMoveIds)).
+			Returning("runs.id").
+			Model(&acceptedIDs).
+			Exec(ctx); err != nil {
+			return nil, fmt.Errorf("updating run's project IDs: %w", err)
+		}
+
+		for _, acceptID := range acceptedIDs {
+			results = append(results, &apiv1.RunActionResult{
+				Error: "",
+				Id:    acceptID,
+			})
+		}
+		if err = tx.Commit(); err != nil {
+			return nil, err
+		}
+	}
+	return &apiv1.MoveRunsResponse{Results: results}, nil
+}
+
+func cloneExperimentAndRun(ctx context.Context, expID int32, runID int32, destProjID int32) error {
+	var cloneExpID int32
+
+	// clone experiment into dest project
+	err := db.Bun().NewRaw(`INSERT INTO experiments(state, config, model_definition, start_time, end_time, archived,
+		parent_id, owner_id, progress, original_config, notes, job_id, project_id, checkpoint_size, checkpoint_count,
+		best_trial_id, unmanaged, external_experiment_id) 
+		SELECT 'COMPLETED'::experiment_state as state, config, model_definition, start_time,
+		(CASE WHEN end_time IS NULL THEN NOW() ELSE end_time END) as end_time, archived, parent_id, owner_id, progress,
+		original_config, notes, job_id, ? as project_id, checkpoint_size, checkpoint_count, best_trial_id, unmanaged,
+		external_experiment_id FROM experiments WHERE id = ? RETURNING id;`, destProjID, expID).Scan(ctx, &cloneExpID)
+	if err != nil {
+		return err
+	}
+
+	var cloneRunID int32
+	// clone run into dest project
+	err = db.Bun().NewRaw(`INSERT INTO runs(experiment_id, state, start_time, end_time, hparams, warm_start_checkpoint_id,
+		best_validation_id, runner_state, restart_id, restarts, tags, checkpoint_size, checkpoint_count,
+		searcher_metric_value, total_batches, searcher_metric_value_signed, latest_validation_id, summary_metrics,
+		summary_metrics_timestamp, last_activity, external_run_id, project_id)
+		SELECT ? as experiment_id,
+		(CASE WHEN state IN ('CANCELED', 'COMPLETED', 'ERROR') THEN state ELSE 'COMPLETED'::trial_state END) as state,
+		start_time,
+		(CASE WHEN end_time IS NULL THEN NOW() ELSE end_time END) as end_time,
+		hparams, warm_start_checkpoint_id, best_validation_id, runner_state, restart_id, restarts, tags, checkpoint_size,
+		checkpoint_count, searcher_metric_value, total_batches, searcher_metric_value_signed, latest_validation_id,
+		summary_metrics, summary_metrics_timestamp, last_activity, external_run_id, ? as project_id
+		FROM runs WHERE id = ? RETURNING id`,
+		cloneExpID, destProjID, runID).Scan(ctx, &cloneRunID)
+	if err != nil {
+		// Delete cloned experiment if run cloning fails
+		err = db.Bun().NewDelete().Table("experiments").Where("id = ?", cloneExpID).Scan(ctx)
+		return err
```

Comment:
we could swallow the error from ```clone run into dest project``` if the delete on the experiments table succeeded. 

Callers would think we did create a new experiment when we didn't

Reply:
updated to show both error on delete error